src/olympia/users/tasks.py

```diff
@@ -196,3 +199,53 @@ def send_suppressed_email_confirmation(suppressed_email_verification_id):
     )
 
     verification.save()
+
+
+@task(
+    autoretry_for=(OperationalError, InterfaceError), max_retries=5, retry_backoff=True
+)
+def bulk_add_disposable_email_domains(entries: list[tuple[str, str]], batch_size=1000):
+    task_log.info(f'Adding {len(entries)} disposable email domains')
+
+    records = []
+    errors = []
+
+    for entry in entries:
+        [domain, provider] = entry
+        record = DisposableEmailDomainRestriction(
+            domain=domain,
+            reason=f'Disposable email domain of {provider}',
+        )
+
+        try:
+            record.full_clean()
+            records.append(record)
+        except ValidationError as e:
+            errors.append(e)
+
+    if not records:
+        task_log.info('No valid entries provided')
+        return
+
+    processed_domains = []
+    records_iter = iter(records)
+
+    if not isinstance(batch_size, int) or batch_size <= 0:
+        raise ValueError('batch_size must be a positive integer')
+
+    while batch := list(itertools.islice(records_iter, batch_size)):
+        created_objects = DisposableEmailDomainRestriction.objects.bulk_create(
+            batch,
+            batch_size,
+            ignore_conflicts=True,
+        )
+        processed_domains.extend(created_objects)
```

Comment:
I've been scratching my head over this for a while - `records` is a `list`... that we've wrapped in an `iter`... that we're slicing with `islice`... that we're casting back to a `list`.  

Just slice `records` directly with something like
```
for i in range(0, len(records), batch_size):
    batch = records[i : i + batch_size]
    ...
```
?

Reply:
And, if we didn't need the separate `task_log.info` calls (do we _need_ them?) then we don't have to manually batch anyway, because setting `batch_size` does the batching inside `bulk_create` for us.

Reply:
Referencing previous comment, I think with these commands/tasks it is better to over log at least in the beginning. Could you give an example of what you are suggesting in terms of changes?

Reply:
Cool. I've left the logs for reasons stated above. Resolving.